controllers/deactivation/deactivation_controller.go

```diff
@@ -278,13 +281,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	return reconcile.Result{}, nil
 }
 
-func (r *Reconciler) resetDeactivatingState(ctx context.Context, usersignup *toolchainv1alpha1.UserSignup) error {
-	if states.Deactivating(usersignup) {
-		states.SetDeactivating(usersignup, false)
-		if err := r.Client.Update(ctx, usersignup); err != nil {
-			log.FromContext(ctx).Error(err, "failed to reset usersignup deactivating state")
-			return err
+func (r *Reconciler) resetDeactivatingState(ctx context.Context, userSignup *toolchainv1alpha1.UserSignup) (*toolchainv1alpha1.UserSignup, error) {
+	if states.Deactivating(userSignup) {
+		states.SetDeactivating(userSignup, false)
+		if err := r.Client.Update(ctx, userSignup); err != nil {
+			log.FromContext(ctx).Error(err, "failed to reset userSignup deactivating state")
+			return userSignup, err
+		}
+		var reloaded *toolchainv1alpha1.UserSignup
+		// Reload the UserSignup
+		if err := r.Client.Get(ctx, types.NamespacedName{
+			Namespace: userSignup.Namespace,
+			Name:      userSignup.Name,
+		}, reloaded); err != nil {
+			// If there was an error, return the existing UserSignup and the error
+			return userSignup, err
 		}
```

Comment:
The original UserSignup object already has the values set, there is no need to reload what you have already set there.
We should rather really return here and reconcile again as it was before.

Reply:
Agreed, have now refactored to work this way instead.